File: eBarbershop.Model/Korisnik.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eBarbershop.Model
{
    public class Korisnik
    {
        public int KorisnikId { get; set; }
        public string Ime { get; set; }
        public string Prezime { get; set; }
        public string Email { get; set; }
        public string Username { get; set; }
        public string Slika { get; set; }

        public int GradId { get; set; }
        public List<string> Uloge => KorisnikUlogas?
                .Select(x => x.Uloga?.Naziv)
                .Where(naziv => !string.IsNullOrEmpty(naziv))
                .ToList(); public virtual ICollection<KorisnikUloga> KorisnikUlogas { get; set; } = new List<KorisnikUloga>();
    }
    
}

File: UI/ebarbershop_mobile/lib/models/korisnik.dart
// korisnik.dart
import 'package:ebarbershop_mobile/models/korisnik_uloga.dart';
import 'package:json_annotation/json_annotation.dart';

part 'korisnik.g.dart';

@JsonSerializable()
class Korisnik {
  int? korisnikId;
  String? ime;
  String? prezime;
  String? email;
  String? username;
  String? slika;
  int? gradId;
  String? uloge;
  List<KorisnikUloga>? korisnikUlogas;

  Korisnik({
    this.korisnikId,
    this.ime,
    this.prezime,
    this.email,
    this.username,
    this.slika,
    this.gradId,
    this.uloge,
    this.korisnikUlogas,
  });

  factory Korisnik.fromJson(Map<String, dynamic> json) => _$KorisnikFromJson(json);
  Map<String, dynamic> toJson() => _$KorisnikToJson(this);

  String get fullName => '$ime $prezime';
  bool get isEmployee => korisnikUlogas?.any((uloga) => uloga.uloga?.naziv == 'Uposlenik') ?? false;
}

